### PR TITLE
drivers/pn532: adapt to new i2c API (recreated)

### PR DESCRIPTION
**Pull request by MichelRottleuthner**
### Contribution description
clone of #9432 - with all the changes addressed. Accidentally dleted the old branch and wasn't able to reopen the PR.

### Issues/PRs references
#6577

### CODE REVIEW

**Comment by dylad** [comment]
@MichelRottleuthner Travis is complaining, there are some styles issues.
You can fix and squash directly.

**Comment by dylad** [comment]
@MichelRottleuthner I have no idea... maybe @kaspar030 has an idea ?
btw, Travis is complaining again with whitespace issue.

**Comment by kaspar030** [comment]
> why are we using such an old version of cppcheck?

Murdock uses the ubuntu 16.04 version, 1.72. Travis seems to be even older. Maybe that can be updated? @aabadie?

**Comment by dylad** [comment]
Yeah, it would be great to update it so we don't have to add voodoo things here.

**Comment by aabadie** [comment]
> Maybe that can be updated?

I prefer dropping Travis and switch to CircleCI like I did in #8729. The advantage is that it uses the official docker image riotdocker so the tools versions will be same has the ones on Murdock.

**Comment by dylad** [comment]
> I prefer dropping Travis and switch to CircleCI like I did in #8729

How do it works for this PR ?

**Comment by aabadie** [comment]
> How do it works for this PR ?

Revert the changes related to cppcheck and only keep the changes related to I2C. If Travis fails, it won't block the merge button anyway.

**Comment by MichelRottleuthner** [comment]
reverted cpp changes. Is the upgrade to the CI planned before the i2c rework branch will be merged?

**Comment by dylad** [comment]
@aabadie you mean we can merge it right now without taking care of Travis ?

**Comment by aabadie** [comment]
>  you mean we can merge it right now without taking care of Travis ?

Yes

**Comment by dylad** [comment]
Let's go then !

File: drivers/pn532/pn532.c [hunk 1]
```diff
@@ -80,7 +80,7 @@
 
 #if ENABLE_DEBUG
 #define PRINTBUFF printbuff
-static void printbuff(char *buff, unsigned len)
+static void printbuff(uint8_t *buff, unsigned len)
```
Comment (aabadie) [inline drivers/pn532/pn532.c]: Is this a leftover of changes related to cppcheck ?

File: drivers/pn532/pn532.c [hunk 2]
```diff
@@ -120,15 +120,7 @@ int pn532_init(pn532_t *dev, const pn532_params_t *params, pn532_mode_t mode)
...
-    if (mode == PN532_I2C) {
-#ifdef PN532_SUPPORT_I2C
-        if (i2c_init_master(dev->conf->i2c, I2C_SPEED_FAST) != 0) {
-            DEBUG("pn532: initialization of I2C bus failed\n");
-            return -1;
-        }
-#endif
-    }
-    else {
+    if (mode == PN532_SPI) {
```
Comment (aabadie) [inline drivers/pn532/pn532.c]: It should be `PN532_I2C`, no ?

Comment (aabadie) [reply drivers/pn532/pn532.c]: Ok, I see

File: drivers/pn532/pn532.c (hunk 1 above)
Comment (aabadie) [reply drivers/pn532/pn532.c]: I would also revert this and open a separate PR. A least, it should not be part of the same commit containing I2C changes

**Comment by dylad** [comment]
@aabadie I'll let you push the button then :)

File: drivers/pn532/pn532.c (hunk 1 above)
Comment (MichelRottleuthner) [reply drivers/pn532/pn532.c]: reverted. shall I open the other PR against new_i2c_if too? (master should work though)

Comment (aabadie) [reply drivers/pn532/pn532.c]: I think it should be fine against master.

**Pull request merged**